portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -80,6 +104,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
-    response.sendRedirect("/comments.html");
+    String url = "/comments.html?numComments=" + numDisplay + "&commentSort=" + sortOrder;
+
+    response.sendRedirect(url);
+  }
+
+  /**
+   * Gets the sort order from the request header, and parses it accordingly.
+   * @return an integer indicating the sort order.
+   */
+  private int getSortOrder(HttpServletRequest request) {
+    String data = request.getHeader("commentSort");
+    if ("timeAscending".equals(data)) {
+      return TIME_ASCENDING;
+    } else if ("timeDescending".equals(data)) {
+      return TIME_DESCENDING;
+    } else {
+      return -1;
+    }
+  }
+
+  /**
+   * @return true if the fields need to be updated, false otherwise
+   */
+  private boolean getToChange(HttpServletRequest request) {
+    String data = request.getHeader("change");
+    return "true".equals(data);
```

Comment:
[Nit] Another Yoda condition

Reply:
Refer to what I said above ^^

Reply:
Hmm, another alternative would be 
`return data != null and data.equals("true")`
This works because 'and' short circuits in javascript.

I think this is my preferred alternative but given that it is a bit longer than your alternative I'll leave it up to you.

Reply:
As a side note, if you end up using the solution above and need to use that pattern multiple times you probably want to factor it out into a function

safeStringEquals(String a, String b) {
  return a != null and b != null and a ==b;
}

Also, I said 'and' short circuits in javascript above, while this is a java file but the same it true in both languages.  This is actually a poor design of these language to begin with.  There's tons written about the fact that any object in Java can be null and why this is bad language design. 
 They tried to fix it in Java 8 and failed, you can see a quick overview of the topic here: https://dzone.com/articles/whats-wrong-java-8-part-iv).  In short, the fact that code is ugly is problem with Java itself, not really a problem with your code.

Reply:
Oh right, I forgot about the short-circuiting && statements. Thanks!